### ISSUE TITLE
[Request] Server ruRO

### ISSUE THREAD

**Issue by sigerraf** [OpenKore/openkore#2181]
## Feature Request 
Implement ruRO access in openkore

###### Affected configuration(s)/ file(s):
```Network/server files```

###### Benefits:
connect to russian Ragnarok Online Server

###### Additional Details:
https://ru.4game.com/ro/play/

server is in closed beta, i can give access to server throught teamviewer or give input data

**Comment by sctnightcore** [comment]
Maybe Wait @kLabMouse

**Comment by alisonrag** [comment]
please be direct in your question, give details, what server is? ragnarok have a lot of official and private server.
give us link, name, test or i will close this issue

**Comment by sigerraf** [comment]
https://ru.4game.com/ro/play/

**Comment by alisonrag** [comment]
days ago someonefrom dev discord:
```
full client can be downloaded by tool from this url
also you cant register in game. only who bought some items before have access to closed beta test
this mean best what you can get is installed client
but cant connect to game
```

```
actually in closed beta test look like no gameguard. or it located outside of game exe/dlls
```

**Comment by sigerraf** [comment]
I can grant access through TeamViewer

**Comment by sigerraf** [comment]
or place the client in the cloud
and give input data

**Comment by kLabMouse** [comment]
I'm already retired from RO. But I can say that Nothing on the communication level will change, as the game is developed by one single company.
On the other hand, there can be additions like packet encryption, some packet variations and so on.

If it is only official RU server available, let it be. If not, then must be as a variation of something else.

**Comment by sctnightcore** [comment]
@kLabMouse  Can u join iRC openkore ?

**Comment by sigerraf** [comment]
official RU server

**Comment by kisti92** [comment]
I myself wanted to write this topic myself. Request for support of this Russian server. This server will be official in Russia, as reported by the company gravity, if Innova (ru) will fulfill the promises and agreements on its part.

**Comment by sigerraf** [comment]
<clientinfo>

    <servicetype>russia</servicetype>
    <servertype>primary</servertype>
    <passwordencrypt></passwordencrypt>
    <hideaccountlist></hideaccountlist>

    <connection>
        <display>Russia Server</display>
        <desc>None</desc>
        <address>109.105.140.144</address>
        <port>6800</port>
        <version>1</version>
        <registrationweb>https://ru.4game.com/</registrationweb>
        <aid>

**Comment by kisti92** [comment]
Yes, I understand that sooner or later I made a mistake. This is from the servers file.

### [Innova]
ip 109.105.140.144
port 6800
master_version 9
version 11001
serverType kRO_RagexeRE_2017_12_27a
addTableFolders Innova
charBlockSize 147
pinCode 1
clientHash 01000000424B33384B57484852000000

**Comment by sigerraf** [comment]
storageEncryptKey 0x000000A1,0x000000B6,0x000000B7,0x000000B8,0x000000B9,0x000000BA,0x000000BB,0x000001E3
login_packet 760a010000004f4b364e504b594600000000170100000b010000160100007a0000007a0000007b0000000b010000acd1fd34ce855e45ee44b5fc4d228bc4284d8f6a7c6936f6bd27d75d133265611900

**Comment by sctnightcore** [comment]
work with xkore1 :p

**Comment by sigerraf** [comment]
Plz pm konfig:-)

**Comment by Angelofline** [comment]
[RagGame]
ip 109.105.140.144
port 6800
master_version 00
version 1
serverType 0
serverEncoding Russian

And after than write сan't take file recvpackets.txt.
Plz help ...)

**Comment by sctnightcore** [comment]
![image](https://user-images.githubusercontent.com/23263315/43073011-40c83d12-8ea2-11e8-8891-5482fc310aaf.png)
:p

**Comment by Angelofline** [comment]
Can you give me config? )--Отправлено из мобильной Яндекс.Почты23.07.2018, 20:00, "Sctnightcore" <notifications@github.com>:
:p

**Comment by sctnightcore** [comment]
Now rRO dont use antibot . U can use xkore1

**Comment by kisti92** [comment]
Respected! Please give info from the file system.

**Comment by alisonrag** [comment]
@sctnightcore will you implement ruRO server in openkore?

i will work with FF to implement code to encrypt password

**Comment by alisonrag** [comment]
@ everyone else

is not simple make openkore work in ruRO

have to add new servertype, recvpackets, shuffle, sync files and update xkore to work with the client

**Comment by Angelofline** [comment]
it remains only to find how to patch it )))

**Comment by Morrikone** [comment]
@sctnightcore   
https://gist.github.com/sctnightcore/38a1a55e1a2b37a199e27e815854256b
It works 15 minutes, then I get error messages
https://cdn1.savepice.ru/uploads/2018/7/24/c85988e5b5504526c3ecbfc53ba0e0a1-full.png
https://cdn1.savepice.ru/uploads/2018/7/24/eb6560817ccee07b6878947705a429fa-full.png
https://cdn1.savepice.ru/uploads/2018/7/24/29f0ef7321192e94e24387a555a4c75a-full.png
https://cdn1.savepice.ru/uploads/2018/7/24/8ea2feeacf280f2211a49edb3178a4f3-full.png

**Comment by alisonrag** [comment]
openkore is spliting the packet wrong, maybe have conflict in recvpackets

**Comment by Morrikone** [comment]
@alisonrag  @sctnightcore  I am very weak in this, unfortunately (
Is it possible to fix everything so that it works? We can discuss the money for it  =/

**Comment by alisonrag** [comment]
@Morrikone 
this server is in closed beta, is better wait open to everyone

someone told to us that the company that is hosting ruRO have their own anti cheat and maybe is testing and trying to implement in client.

**Comment by Morrikone** [comment]
@alisonrag Well, thank you. Protection will be completed in a closed beta i hope you can do something later

**Comment by alisonrag** [comment]
i hope also =]

this is why we are not moving so fast now.

for now you can use for a limited time, but this is not the launch version, so we prefer to wait

**Comment by kisti92** [comment]
@Morrikone If you still managed to start. Even for 30 minutes. Could you share your files?

**Comment by sigerraf** [comment]
write Мясо, Белое зелье :-)

**Comment by kisti92** [comment]
Comrades those who got it started and that this script (bot) was working. I understand that no one is not going to share for free. :( I'm ready to pay a certain amount in order to set up my teamviewer for example.

**Comment by Morrikone** [comment]
@kisti92 Успокойся ;) После релиза люди будут тестировать.

**Comment by sigerraf** [comment]
Щас просто проверка скриптов.я проверил чуть подкорректировал и выключил

**Comment by kisti92** [comment]
Of course, I understand everything. As they said "the bot release will be on the server, after the open access". But this access may not appear soon ... can you still do it?

**Comment by alisonrag** [comment]
can you post download link?

**Comment by kisti92** [comment]
@alisonrag 
If you need a link to download the game server. Then I'll upset you, download only those who are allowed to close testing for $ 10. But, I can provide my account if this helps setting up the bot.
Only how can I give you the account information anonymously?

**Comment by sigerraf** [comment]
Напиши мне на почту я тебе помогу настроить 2icarus@inbox.ru

**Comment by kisti92** [comment]
@alisonrag 
Luckily a good man is a **sigerraf**. He helped me to run. Now I will try to adjust, as I need.

**Comment by Pu3Ho** [comment]
@kisti92 За 600 рублей тебе "помог"?

**Comment by sigerraf** [comment]
а ты автору перевел хоть копейку "робингуд"?

**Comment by Pu3Ho** [comment]
@sigerraf Он где то оставлял адрес куда переводить? Вроде нет, да и софт бесплатный и делаеться на добровольной основе. Да и патч, он сделал/выложил, бесплатно/безвозмездно.

**Comment by sigerraf** [comment]
Нажми на его профиль "одарённый"

**Comment by Pu3Ho** [comment]
@kisti92 А, он тебе сказал что с более чем 99% вероятностью, они допилят интеграцию фроста в клиент к моменту релиза и при инжекте опенкора в клиент, в Xkore 1 моде, фрост автоматически выпишет тебе бан в течении 10 минут(ну, а если же нет - экономика на серваке будет убита за месяц)?

**Comment by kisti92** [comment]
@Pu3Ho 
Ну я надеюсь, что над обходом будут работать. 
Странный вопрос про экономику на сервере и то что она будет убита. Так как этим ты выражаешь не согласие использования ботов на данном сервер. (Что ты тут делаешь тогда?:) )

**Comment by sigerraf** [comment]
А я просто понимаю что человек решил пофлудить в силу горящего пукана. Сам то что ты сделал для настройки коры? Или помог кому-то ? Я думаю не стоит отвечать на письмена данного оппонента дабы незафлуживать тему и ждать завтрашней профы

**Comment by sigerraf** [comment]
И решать проблему по мере ее поступления

**Comment by Pu3Ho** [comment]
@kisti92 Скажем так, в идеале : я больше за то что бы ботов там не было(ну хотя бы, первое время), ибо хочется поиграть в РО в таком виде, как он задуман. Но, если они будут.... я в стороне не останусь, ибо зачем быть крайним :)

@sigerraf Уже как минимум указал на фрост, скинул автору патчей 10$, вместо того чтобы кидать их тебе, и еще кое что чуть ниже :)

Here's the Openkore with @sctnightcore's patches applied for RuRo/Опенкор с патчами для 4game : https://ufile.io/1u6zd

**Comment by sigerraf** [comment]
Сам радеет за отсутствие ботов на сервере и выкладывает прямую ссылку на бота)))

**Comment by Pu3Ho** [comment]
@sigerraf Мне пофиг на ботов на ЗБТ, а на релизе он всё равно работать не будет + ты не будешь его продавать(либо "клиентов" будет меньше).

**Comment by Morrikone** [comment]
@sigerraf Щас бы продать бесплатный продукт который вымолил. =( Сиди помалкивай спокойно в тряпочку.

**Comment by sctnightcore** [comment]
english :p

**Comment by sctnightcore** [comment]
miss click
